pkg/limits/ingest_limits.go

```diff
@@ -106,7 +106,7 @@ type IngestLimits struct {
 	limits Limits
 
 	// Track stream metadata
-	metadata StreamMetadata
+	metadata StreamMetadataStore
```

Comment:
nit: It looks like there's only one implementation of this interface. How likely is it that we'll have more? If it's unlikely, should `MemStreamMetadataStore` become `StreamMetadataStore`? 

(IMO interfaces add some mental indirection, if we only ever have one implementation of an interface it can become cognitively noisy)

Reply:
I think unlikely, so I am going to remove the interface and make it a concrete type 👍